pyccel/parser/semantic.py

```diff
@@ -1108,11 +1148,36 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign,ar
             if not arr_in_multirets:
                 self._ensure_target(rhs, d_lhs)
 
-            var = self.check_for_variable(name)
-
+            if isinstance(lhs, DottedName):
+                var = self.check_for_attribute(lhs)
+            else:
+                var = self.check_for_variable(name)
             # Variable not yet declared (hence array not yet allocated)
             if var is None:
-
+                if isinstance(lhs, DottedName):
+                    if (self._current_function == '__init__' or self.current_class) and lhs.name[0] == 'self':
+                        cls      = self.get_variable('self')
+                        cls_name = str(cls.cls_base.name)
+                        cls      = self.scope.find(cls_name, 'classes')
+
+                        attributes = cls.attributes
+                        parent     = cls.superclasses
+                        attributes = list(attributes)
+                        name     = str(lhs.name[-1])
+
+                        var      = self.get_variable('self')
+
+                        member = self._create_variable(name, dtype, rhs, d_lhs)
+                        lhs    = member.clone(member.name, new_class = DottedVariable, lhs = var)
+
+                        # update the attributes of the class and push it to the scope
+                        attributes += [member]
+                        new_cls = ClassDef(cls_name, attributes, [], superclasses=parent)
+                        self._current_class = new_cls
+                        self.scope.parent_scope.update_class(new_cls)
```

Comment:
Do you think we can update the existing class or do we need to create a new one each time? Shouldn't your setter let us simplify this?